### DISCUSSION TITLE
Auro Flight Element feature request

### DISCUSSION THREAD

**Discussion by blackfalcon** [AlaskaAirlines/WC-Generator#531] [Announcements]
Submitted by Robert Hays

We are using the Auro Flight element within the RAIN application.  On occasion, we receive invalid flight departure/arrival datetime values from GDS/Sabre services.  

For example, we might receive a minimum value datetime such as `0001-01-01T12:00:00+08:00` for either the departure or arrival date.  In this situation (shown in the screen shot), you would see a `12:00 pm` value displayed in the Auro Flight element for an arrival datetime value of `0001-01-01T12:00:00+08:00`.  This is technically a valid date so we will not see the 'Invalid Date' that would typically display in the Auro Flight element when it is truly an invalid date situation.  

Our business customers would prefer to display some sort of custom text (maybe something like 'Time Not Available') vs. the `12:00 pm` value that we current display.  Is there a way to override or provide a customizable departure/arrival time value/description within the Auro Flight element?

![image](https://github.com/AlaskaAirlines/WC-Generator/assets/181089/77e2c718-3740-4e72-bc0c-b0c4cac114d0)

**Comment by blackfalcon** [comment]
You are correct, `0001-01-01T12:00:00+08:00` is technically a valid value and the `auro-flight` element can extract the time code out of that value. 

Since this is a valid time code and you say 'we **might** receive a minimum value datetime such as `0001-01-01T12:00:00+08:00`', what other bad time codes do you receive? 

We could consider a check to validate of the time code returns the current year or maybe the last 10?

**Reply by RHays70** [reply]
Thanks for your response.  Right now we are only seeing this specific issue where the min date value of '0001-01-01T12:00:00+08:00' is coming from GDS/Sabre and mapped to the Auro flight element in RAIN.  We are working to better understand the issue with the teams that support these services.  I suppose there could be other similar date related issues but this min date issue is the only one that I'm aware of at the moment.  We recently identified a bug within RAIN UI where we were creating invalid dates that were used in the Auro flight element by means of some time zone offset functionality.  I have a bug to fix that issue.

**Reply by blackfalcon** [reply]
Ok, this seems fair. I will write up a new feature to this specification. 

Update the component's `departureTime` and `arrivalTime` to sanitize the incoming time data and eval if year is within the last '10' year(s). 

If eval returns false, update UI to reflect not a valid time. 

Please let me know if there is anything I am missing.

**Reply by RHays70** [reply]
Thanks so much for adding this new feature.  The recommendation from the RAIN Product/Design team would be to display "Time Unavailable" for this scenario.  I've also been asked about when this functionality might be available so if you have a estimate on that then please advise.  Appreciate you taking this into consideration.
